src/managers/GuildMemberRoleManager.js

```diff
@@ -100,17 +100,23 @@ class GuildMemberRoleManager {
    */
   async add(roleOrRoles, reason) {
     if (roleOrRoles instanceof Collection || Array.isArray(roleOrRoles)) {
-      roleOrRoles = roleOrRoles.map(r => this.guild.roles.resolve(r));
-      if (roleOrRoles.includes(null)) {
-        throw new TypeError('INVALID_TYPE', 'roles', 'Array or Collection of Roles or Snowflakes', true);
+      const resolvedRoles = [];
+      for (const role of roleOrRoles instanceof Collection ? roleOrRoles.values() : roleOrRoles) {
+        const resolvedRole = this.guild.roles.resolve(role);
+        if (!resolvedRole) {
+          return Promise.reject(new TypeError('INVALID_ELEMENT', 'Array or Collection', 'roles', role));
+        }
+        resolvedRoles.push(resolvedRole);
       }
 
-      const newRoles = [...new Set(roleOrRoles.concat(...this._roles.values()))];
+      const newRoles = [...new Set(resolvedRoles.concat(...this._roles.values()))];
       return this.set(newRoles, reason);
     } else {
       roleOrRoles = this.guild.roles.resolve(roleOrRoles);
       if (roleOrRoles === null) {
```

Comment:
```suggestion
      roleOrRoles = this.guild.roles.resolveID(roleOrRoles);
      if (!roleOrRoles) {
```
Since `roleOrRoles` is the ID now, this requires some changes below